src/main/java/at/ac/tuwien/ifs/sge/leeroy/mcts/AttackMctsActionSupplier.java

```diff
@@ -0,0 +1,152 @@
+package at.ac.tuwien.ifs.sge.leeroy.mcts;
+
+import at.ac.tuwien.ifs.sge.game.risk.board.Risk;
+import at.ac.tuwien.ifs.sge.game.risk.board.RiskAction;
+import at.ac.tuwien.ifs.sge.game.risk.board.RiskBoard;
+import at.ac.tuwien.ifs.sge.leeroy.agents.AttackActionSupplier;
+import at.ac.tuwien.ifs.sge.leeroy.agents.GameUtils;
+import at.ac.tuwien.ifs.sge.leeroy.agents.OccupyActionSupplier;
+import at.ac.tuwien.ifs.sge.util.Util;
+
+import java.util.*;
+import java.util.function.BooleanSupplier;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class AttackMctsActionSupplier extends MctsActionSupplier{
+
+    /**
+     * magic numbers for optimal action selection
+     */
+    private static final double OCCUPATION_FACTOR = 20; // number of our occupied territories
+    private static final double FRONTLINE_PENALTY_FACTOR = -.25; // number of enemy territories on the frontline
+    private static final double FRONTLINE_MARGIN_FACTOR = 0.75; // margin of frontline troops (ours minus theirs)
+    private static final double CONTINENT_BONUS_FACTOR = 100; // bonus for occupied continents
+    private static final double ENEMY_CONTINENT_PENALTY_FACTOR = -40; // malus for enemy occupied continents
+    private static final double UNUSED_TROOPS_PENALTY_FACTOR = -0.5; // malus for inefficient attack/occupy
+
+    private final static int MAX_ATTACK_TROOPS = 3;
+
+    public AttackMctsActionSupplier(BooleanSupplier shouldStopComputation) {
+        super(shouldStopComputation);
+    }
+
+    /**
+     * first expand territories with max defending troops -> attacker looses less troops -> maximize expected value
```

Comment:
Why does the attacker lose the fewest troops when the defender has the most defending troops?

Reply:
please check https://boardgames.stackexchange.com/a/20509 for an example

Reply:
Ah ok, I did not understand this was in regard to the strategy, thank you!